packages/js/src/generators/release-version/release-version.ts

```diff
@@ -434,6 +441,85 @@ To fix this you will either need to add a package.json file at that location, or
             }
             break;
           }
+          case 'version-plans': {
+            if (!options.releaseGroup.versionPlans) {
+              if (
+                options.releaseGroup.name === IMPLICIT_DEFAULT_RELEASE_GROUP
+              ) {
+                throw new Error(
+                  `Invalid specifierSource "version-plans" provided. To enable version plans, set the "release.versionPlans" configuration option to "true" in nx.json.`
+                );
+              } else {
+                throw new Error(
+                  `Invalid specifierSource "version-plans" provided. To enable version plans for release group "${options.releaseGroup.name}", set the "versionPlans" configuration option to "true" within the release group configuration in nx.json.`
+                );
+              }
+            }
+
+            if (options.releaseGroup.projectsRelationship === 'independent') {
+              specifier = (
+                options.releaseGroup.versionPlans as ProjectsVersionPlan[]
+              ).reduce((spec: ReleaseType, plan: ProjectsVersionPlan) => {
+                if (!spec) {
+                  return plan.projectVersionBumps[projectName];
+                }
+                if (plan.projectVersionBumps[projectName]) {
+                  const prevNewVersion = inc(currentVersion, spec);
+                  const nextNewVersion = inc(
+                    currentVersion,
+                    plan.projectVersionBumps[projectName]
+                  );
+                  return gt(nextNewVersion, prevNewVersion)
+                    ? plan.projectVersionBumps[projectName]
+                    : spec;
+                }
+                return spec;
+              }, null);
+            } else {
+              specifier = (
+                options.releaseGroup.versionPlans as GroupVersionPlan[]
+              ).reduce((spec: ReleaseType, plan: GroupVersionPlan) => {
+                if (!spec) {
+                  return plan.groupVersionBump;
+                }
+
+                const prevNewVersion = inc(currentVersion, spec);
+                const nextNewVersion = inc(
+                  currentVersion,
+                  plan.groupVersionBump
+                );
+                return gt(nextNewVersion, prevNewVersion)
+                  ? plan.groupVersionBump
+                  : spec;
+              }, null);
+            }
+
+            if (!specifier) {
+              specifier = null;
+              log(`🚫 No changes were detected within version plans.`);
```

Comment:
```suggestion
            if (!specifier) {
                if (projectToDependencyBumps.has(projectName)) {
                    // No applicable changes to the project directly by the user, but one or more dependencies have been bumped and updateDependents is enabled
                    specifier = updateDependentsBump;
                    log(`📄 Resolved the specifier as "${specifier}" because "release.version.generatorOptions.updateDependents" is enabled`);
                } else {
                    specifier = null;
                    log(`🚫 No changes were detected within version plans.`);
                }
            }
```